src/script/cryptography/CryptographyMapper.js

```diff
@@ -186,15 +186,17 @@ export default class CryptographyMapper {
       data.info.tag = 'medium';
     }
 
-    if (uploaded !== null) {
+    if (asset.hasOwnProperty('uploaded') && uploaded !== null) {
       data = Object.assign(data, {
         key: uploaded.assetId,
         otr_key: new Uint8Array(uploaded.otrKey),
         sha256: new Uint8Array(uploaded.sha256),
         status: z.assets.AssetTransferState.UPLOADED,
         token: uploaded.assetToken,
       });
-    } else {
+    }
+
+    if (asset.hasOwnProperty('notUploaded') && notUploaded !== null) {
```

Comment:
else if seems appropriate here

Reply:
Agreed

Reply:
Seems appropriate, but isn't :smiley: it was changed in the last refactor, now I am changing it back, see https://github.com/wireapp/wire-webapp/pull/5831/files#diff-d2363003337179b68a0f317d6da77c58L198

Reply:
Ok, @atomrc conviced me.